src/display/base_factory.js

```diff
@@ -46,10 +46,13 @@ class BaseFilterFactory {
 }
 
 class BaseCanvasFactory {
-  constructor() {
+  #enableHardwareAcceleration = false;
+
+  constructor(enableHardwareAcceleration = false) {
```

Comment:
Please keep the constructor signatures consistent in the base- and extending-classes.
```suggestion
  constructor({ enableHardwareAcceleration = false }) {
```